src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -105,6 +111,38 @@ protected function formatValue($value, $format = 0)
             return $value->format('Y-m-d H:i:s');
         }
 
+        if (($format & self::PRETTY_DATE_INTERVAL) && $value instanceof \DateInterval) {
+            $formattedValueParts = [];
+            foreach ([
+                'y' => 'year',
+                'm' => 'month',
+                'd' => 'day',
+                'h' => 'hour',
+                'i' => 'minute',
+                's' => 'second',
+                'f' => 'microsecond',
+            ] as $p => $label) {
+                if (!($formattedValuePart = $value->format('%'.$p))) {
+                    continue;
+                }
+
+                $formattedValuePart .= ' '.$label;
```

Comment:
Maybe
`
$formattedValuePart .= ' '.$label.($formattedValuePart > 0 ? 's' : '');
`
would be more clearly? Comparison of character string and a number is a bit confusing.

Reply:
Thanks, that looks like a mistake indeed 😅